File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItLinkedBrokerData.java
package it.trade.android.sdk.model;

import java.util.ArrayList;
import java.util.List;

public class TradeItLinkedBrokerData {

    public String broker;
    public String userId;
    public String userToken;
    public boolean isLinkActivationPending = false;
    public List<TradeItLinkedBrokerAccountData> linkedBrokerAccountDataList = new ArrayList<>();

    public TradeItLinkedBrokerData(String broker, String userId, String userToken) {
        this.broker = broker;
        this.userId = userId;
        this.userToken = userToken;
    }

    public TradeItLinkedBrokerData withLinkActivationPending(boolean isLinkActivationPending) {
        this.isLinkActivationPending = isLinkActivationPending;
        return this;
    }

    public void injectAccount(TradeItLinkedBrokerAccountData linkedBrokerAccountData) {
        this.linkedBrokerAccountDataList.add(linkedBrokerAccountData);
    }

    public TradeItLinkedBrokerData(TradeItLinkedLoginParcelable linkedLoginParcelable) {
        this.broker = linkedLoginParcelable.broker;
        this.userId = linkedLoginParcelable.userId;
        this.userToken = linkedLoginParcelable.userToken;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TradeItLinkedBrokerData that = (TradeItLinkedBrokerData) o;

        return (userId.equals(that.userId) && userToken.equals(that.userToken));
    }

    @Override
    public int hashCode() {
        int result = userId != null ? userId.hashCode() : 0;
        result = 31 * result + (userToken != null ? userToken.hashCode() : 0);
        return result;
    }
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/model/TradeItLinkedBrokerData.kt
package it.trade.android.sdk.model

import java.util.*

class TradeItLinkedBrokerData {

    var broker: String
    var userId: String
    var userToken: String
    var isLinkActivationPending = false
    var linkedBrokerAccounts: MutableList<TradeItLinkedBrokerAccountData> = ArrayList()

    constructor(broker: String, userId: String, userToken: String) {
        this.broker = broker
        this.userId = userId
        this.userToken = userToken
    }

    fun withLinkActivationPending(isLinkActivationPending: Boolean): TradeItLinkedBrokerData {
        this.isLinkActivationPending = isLinkActivationPending
        return this
    }

    fun injectAccount(linkedBrokerAccountData: TradeItLinkedBrokerAccountData) {
        this.linkedBrokerAccounts.add(linkedBrokerAccountData)
    }

    fun injectAccounts(linkedBrokerAccounts: MutableList<TradeItLinkedBrokerAccountData>) {
        this.linkedBrokerAccounts = linkedBrokerAccounts
    }

    constructor(linkedLoginParcelable: TradeItLinkedLoginParcelable) {
        this.broker = linkedLoginParcelable.broker
        this.userId = linkedLoginParcelable.userId
        this.userToken = linkedLoginParcelable.userToken
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val that = other as TradeItLinkedBrokerData?

        return userId == that!!.userId && userToken == that.userToken
    }

    override fun hashCode(): Int {
        var result = userId.hashCode()
        result = 31 * result + userToken.hashCode()
        return result
    }
}
